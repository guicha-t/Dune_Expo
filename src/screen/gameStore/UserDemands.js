import React, { Component } from 'react';
import { Alert, Button, TextInput, View, Text, StyleSheet, ScrollView, Image, TouchableOpacity, AsyncStorage, FlatList, TouchableHighlight, Modal} from 'react-native';
import { observer } from 'mobx-react';
import { List, ListItem } from "react-native-elements";
import GridView from 'react-native-super-grid';

import Header from './../../global/header/Header';
import Store from './../../global/store/Store';
import * as cfg from "./../../Config";

@observer
export default class UserDemands extends Component {
  constructor(props){
    super(props);
    this.state = {
      GamesRequested: [],
      ProfArray:[],
        ModalVisibleStatus: false,
        idAppModal: null,
        nomAppModal: null,
        dateDemandeModal: null,
        commentaireModal: null,
        idToNotify: null,
        idNotif: null
    }

    this.ts = new Date();

    this.contentToRender = []
  }

  componentDidMount(){
   fetch(cfg.API_URL + '/notifs/popUpMenu', {
      method: 'GET',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
        token: Store.Token,
      },
    }).then((response) => response.json())
        .then((responseJson) => {
          this.setState({'GamesRequested':responseJson.response});
        })
        .catch((error) => {
          console.error(error);
        });
  }

  renderProfArray = (param) => {

      fetch(cfg.API_URL + '/notifs/getArrayProf/' + param.toString(), {
          method: 'GET',
          headers: {
              Accept: 'application/json',
              'Content-Type': 'application/json',
              token: Store.Token,
          },
      }).then((response) => response.json())
          .then((responseJson) => {
              this.setState({'ProfArray': responseJson.response});
          })
          .catch((error) => {

              console.error(error);
          });

  }

  readNotification = () => {
      fetch(cfg.API_URL + '/notifs/read/' + this.state.idNotif.toString(), {
          method: 'PUT',
          headers: {
              Accept: 'application/json',
              'Content-Type': 'application/json',
              token: Store.Token,
          }
      }).then((response) => response.json())
          .then((responseJson) => {
          })
          .catch((error) => {
              console.error(error);
          });
  }

  _confirmDemand = () => {
    fetch(cfg.API_URL + '/store/validating', {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
        token: Store.Token,
      },
      body: JSON.stringify({
        idDemande: this.state.idToNotify,
        validate: 1,
      }),
    }).then((response) => response.json())
        .then((responseJson) => {
            this.readNotification();
            this.ShowModalFunction(!this.state.ModalVisibleStatus);
            this.props.navigation.navigate('Dashboard');
    })
    .catch((error) => {
        Alert.alert("ERROR", error);
      console.error(error);
    });
  }


  _cancelDemand = () => {
    fetch(cfg.API_URL + '/store/validating', {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
        token: Store.Token,
      },
      body: JSON.stringify({
        idDemande: this.state.idToNotify,
        validate: '0',
      }),
    }).then((response) => response.json())
        .then((responseJson) => {
            this.readNotification();
            this.ShowModalFunction(!this.state.ModalVisibleStatus);
            this.props.navigation.navigate('Dashboard');
    })
    .catch((error) => {
        Alert.alert("ERROR", error);
      console.error(error);
    });
  }


    ShowProfInModal = (visible, item) =>{

        this.renderProfArray(item.idToNotify);

        this.setState({idToNotify: item.idToNotify.toString()});

        this.setState({ModalVisibleStatus: visible});

        this.setState({idAppModal: item.idApp});

        this.setState({nomAppModal: item.nomApp});

        this.setState({dateDemandeModal: item.dateDemande});

        this.setState({commentaireModal: item.commentaire});

        this.setState({idNotif: item.idNotif});

    }

    ShowModalFunction = (visible) => {

      this.setState({'ModalVisibleStatus': visible});

    }


    render() {
    return (
        <View style={styles.mainContainer}>
          <Header navigation={this.props.navigation}/>
    	<View style={{flex:0.2, alignItems: 'center', justifyContent:'center',}}>
          <Text style={{fontSize:20,}}>
             Application(s) demandée(s)
          </Text>
        </View>
            <View style={{flex: 0.8}}>
                <List>
                    <FlatList
                        data={this.state.GamesRequested}
                        extraData={this.state}
                        rightIcon={'../../picture/profil/eye.png'}
                        renderItem={({ item }) => (
                            <ListItem
                                title={`${ item.nomApp }`}
                                onPress={() => this.ShowProfInModal(true, item)}
                            />
                            )}
                        keyExtractor={item => item.idNotif.toString()}
                    />
                </List>
                <View>
                    <Modal
                        transparent={true}
                        animationType={"slide"}
                        visible={this.state.ModalVisibleStatus}
                        onRequestClose={ () => { this.ShowModalFunction(!this.state.ModalVisibleStatus)} } >
                        <View style={{ flex:1, justifyContent: 'center', alignItems: 'center' }}>
                            <View style={styles.ModalInsideView}>
                                    <Text style={styles.TextStyle}> Vous avez une demande sur l'application
                                            { " " + this.state.nomAppModal }.
                                    </Text>
                                <Text onPress={() => this.props.navigation.navigate('GameContainer', {id: this.state.idAppModal})}  style={{textDecorationLine: 'underline',fontSize: 20, color: "#fff", textAlign: 'center'}}>
                                    Voir l'application
                                </Text>
                                <Text  style={styles.TextProfStyle}> Cette demande a été faite par:</Text>
                                <GridView
                                    itemDimension={100}
                                    spacing={1}
                                    items={this.state.ProfArray}
                                    style={styles.GridView}
                                    renderItem={item => (
                                        <View style={styles.itemContainer}>
                                            <Text>{item.length}</Text>
                                            <TouchableOpacity style={{flex: 1}} onPress={() => {
                                                Alert.alert(item.dateDemande, item.commentaire);
                                            }}>
                                                <View style={{flex: 1, marginLeft: 10, marginRight: 10}}>
                                                    <View style={{flex: 0.7, paddingTop: 10}}>
                                                        <Image
                                                            style={{flex: 1, borderRadius: 10}}
                                                            source={{uri: cfg.API_URL + '/files/profs/' + item.picPath}}
                                                        />
                                                    </View>
                                                    <View style={{flex: 0.3, justifyContent: 'center', alignItems: 'center'}}>
                                                        <Text style={styles.ProfName}>{item.nomPrenom}</Text>
                                                    </View>
                                                </View>
                                            </TouchableOpacity>
                                        </View>
                                    )}
                                />
                                <View style={{flex: 1, flexDirection:'row', justifyContent: 'center', alignItems: 'center', marginTop: '10%', marginBottom: '5%'}}>
                                    <TouchableOpacity style={{margin: '5%'}} onPress={ this._cancelDemand}>
                                        <Image
                                            style={{height: 50, width: 50}}
                                            source={require('./../../picture/profil/errorWhite.png')}
                                            resizeMode="contain"
                                        />
                                    </TouchableOpacity>
                                    <TouchableOpacity style={{margin: '5%'}} onPress={this._confirmDemand}>
                                        <Image
                                            style={{height: 50, width: 50}}
                                            source={require('./../../picture/profil/successWhite.png')}
                                            resizeMode="contain"
                                        />
                                    </TouchableOpacity>
                                </View>
                                <Button title={'Retour'} color='#363453' onPress={() => { this.ShowModalFunction(!this.state.ModalVisibleStatus)} } />
                            </View>
                        </View>
                    </Modal>
                </View>
            </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({


  container: {
   flex: 1,
   paddingTop: 22
  },
  item: {
    padding: 10,
    fontSize: 18,
    height: 44,
  },


  mainContainer: {
    flex:1,
    backgroundColor: '#fff',
  },
  classContainer: {
    flex: 0.1,
    flexDirection: 'row',
    paddingTop: 2,
  },
  searchContainer: {
    flex: 0.1,
    flexDirection: 'row',
    paddingTop: 2,
    alignItems: 'center',
    margin: 5,
  },
  input: {
    flex: 1,
    height: 44,
    padding: 10,
    marginRight: 5,
    borderWidth: 1,
    borderColor: 'black',
  },
  ButtonSearch: {
    padding: 10,
    borderWidth: 1,
    borderColor: 'black',
    marginBottom: 10,
  },
  allClassContainer: {
    flex:0.2,
  },
  gridView: {
    paddingTop: 25,
    flex: 1
  },
  itemContainer: {
    flex: 1,
    height: 130,
    backgroundColor: '#FFF',
    borderRadius:10,
  },
  itemName: {
    fontSize: 18,
    color: '#434B77',
    fontWeight: '600',
  },
  buttonClassAll: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#FEE599',
    margin:2,
  },
  buttonClass: {
    width:178,
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    margin: 2,
    backgroundColor: '#FEE599',
    borderRadius:10,
  },
  textClass: {
    fontSize: 14,
    color: '#434B77',
    fontWeight: '600',
  },
  ProfName: {
    fontSize: 12,
    color: '#434B77',
    fontWeight: '600',
  },
    MainContainer :{

        flex:1,
        justifyContent: 'center',
        alignItems: 'center'

    },

    ModalInsideView:{

        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor : "#363453",
        height: 450 ,
        width: '90%',
        borderRadius:10,
        borderWidth: 1,
        borderColor: '#fff'

    },

    TextStyle:{

        fontSize: 20,
        color: "#fff",
        textAlign: 'center'

    },
    TextProfStyle:{
        fontSize: 20,
        color: "#fff",
        textAlign: 'center',
        padding: 20
    }
});
