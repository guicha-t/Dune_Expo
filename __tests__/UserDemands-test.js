import UserDemands from './../src/screen/gameStore/UserDemands';
import 'isomorphic-fetch';
import React from 'react';

import renderer from 'react-test-renderer'

jest.useFakeTimers();

test('render UserDemands correctly', () => {

    const tree = renderer.create(<UserDemands />).toJSON();
    expect(tree).toMatchSnapshot();
});