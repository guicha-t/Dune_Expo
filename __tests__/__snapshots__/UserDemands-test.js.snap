// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render UserDemands correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFF",
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "borderBottomColor": "black",
          "elevation": 5,
          "flexDirection": "row",
          "height": 56,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.8,
          "shadowRadius": 2,
        },
        Object {
          "backgroundColor": "#f6c6d2",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 0.3,
          "justifyContent": "center",
          "paddingLeft": 10,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 24,
              "width": 24,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 0.4,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 1,
            "opacity": 1,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={1}
          style={
            Object {
              "flex": 1,
              "height": undefined,
              "width": undefined,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 0.3,
          "justifyContent": "center",
          "paddingRight": 10,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={1}
          style={
            Object {
              "height": 32,
              "width": 32,
            }
          }
        />
      </View>
    </View>
    <Modal
      animationType="fade"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      supportedOrientations={
        Array [
          "portrait",
          "portrait-upside-down",
          "landscape",
          "landscape-left",
          "landscape-right",
        ]
      }
      transparent={true}
      visible={false}
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flex": 1,
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#FFF",
              "borderColor": "#ea4335",
              "borderRadius": 5,
              "borderWidth": 1,
              "marginHorizontal": 30,
              "maxWidth": 350,
              "shadowColor": "#000000",
              "shadowOpacity": 0.1,
              "shadowRadius": 10,
              "transform": Array [
                Object {
                  "scale": 0,
                },
              ],
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "justifyContent": "center",
                  "padding": 20,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#333",
                      "fontSize": 18,
                      "fontWeight": "500",
                      "lineHeight": 25,
                      "textAlign": "center",
                    },
                    undefined,
                  ]
                }
              >
                ATTENTION !
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#666",
                      "fontSize": 17,
                      "paddingTop": 10,
                      "textAlign": "center",
                    },
                    undefined,
                  ]
                }
              >
                Vous devez accepter les conditions d'utilisation avant de pouvoir utiliser l'application.
              </Text>
            </View>
            <View
              style={
                Object {
                  "borderRadius": 5,
                  "flexDirection": "row",
                  "flexWrap": "wrap",
                  "justifyContent": "center",
                }
              }
            >
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#ea4335",
                    "borderRadius": 5,
                    "marginBottom": 20,
                    "marginHorizontal": 10,
                    "minWidth": 90,
                    "opacity": 1,
                    "paddingVertical": 8,
                  }
                }
                testID="buttonConfirm"
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#FFF",
                        "fontSize": 15,
                        "fontWeight": "600",
                        "textAlign": "center",
                      },
                      undefined,
                    ]
                  }
                >
                  COMPRIS
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </Modal>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#FFF",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#FFF",
          "marginTop": 15,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "marginLeft": 15,
              "marginRight": 15,
            },
            undefined,
            undefined,
            undefined,
          ]
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={null}
        >
          <View
            pointerEvents="box-only"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#9E9E9E",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 19,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Object {
                  "padding": 12,
                },
                undefined,
                undefined,
                undefined,
                Object {
                  "backgroundColor": "#2A5E85",
                  "borderColor": "white",
                  "borderRadius": 30,
                  "borderWidth": 2,
                  "paddingLeft": 20,
                  "width": 60,
                },
                undefined,
                undefined,
              ]
            }
          >
            <Text />
            <Text
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 20,
                    },
                    undefined,
                    Object {
                      "fontSize": 17.5,
                    },
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFF",
          "flex": 0.2,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#2A5E85",
            "fontSize": 20,
            "textAlign": "center",
          }
        }
      >
        Accepter une demande d'application vous fera utiliser l'un de vos 
        0
         crédit(s) gratuit(s).
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#FFF",
          "flex": 0.8,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#2A5E85",
            "fontSize": 20,
            "paddingTop": 80,
            "textAlign": "center",
          }
        }
      >
         Aucune notification. 
      </Text>
      <View>
        <Modal
          animationType="slide"
          hardwareAccelerated={false}
          onRequestClose={[Function]}
          transparent={true}
          visible={false}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#2A5E85",
                  "borderColor": "#fff",
                  "borderRadius": 10,
                  "borderWidth": 1,
                  "height": 450,
                  "justifyContent": "center",
                  "width": "90%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#fff",
                    "fontSize": 20,
                    "textAlign": "center",
                  }
                }
              >
                 Vous avez une demande sur l'application
                 null
                .
              </Text>
              <Text
                onPress={[Function]}
                style={
                  Object {
                    "color": "#f6c6d2",
                    "fontSize": 20,
                    "textAlign": "center",
                    "textDecorationLine": "underline",
                  }
                }
              >
                Voir l'application
              </Text>
              <Text
                style={
                  Object {
                    "color": "#fff",
                    "fontSize": 20,
                    "padding": 20,
                    "textAlign": "center",
                  }
                }
              >
                 Cette demande a été faite par:
              </Text>
              <RCTScrollView
                data={Array []}
                disableVirtualization={false}
                getItem={[Function]}
                getItemCount={[Function]}
                horizontal={false}
                initialNumToRender={10}
                itemWidth={null}
                keyExtractor={[Function]}
                maxToRenderPerBatch={10}
                numColumns={1}
                onContentSizeChange={[Function]}
                onEndReachedThreshold={2}
                onLayout={[Function]}
                onMomentumScrollEnd={[Function]}
                onScroll={[Function]}
                onScrollBeginDrag={[Function]}
                onScrollEndDrag={[Function]}
                removeClippedSubviews={false}
                renderItem={[Function]}
                scrollEventThrottle={50}
                stickyHeaderIndices={Array []}
                style={
                  Array [
                    Object {
                      "paddingTop": 1,
                    },
                    Object {},
                  ]
                }
                updateCellsBatchingPeriod={50}
                viewabilityConfigCallbackPairs={Array []}
                windowSize={21}
              >
                <View />
              </RCTScrollView>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "marginBottom": "5%",
                    "marginTop": "10%",
                  }
                }
              >
                <View
                  accessible={true}
                  isTVSelectable={true}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "margin": "5%",
                      "opacity": 1,
                    }
                  }
                >
                  <Image
                    resizeMode="contain"
                    source={1}
                    style={
                      Object {
                        "height": 50,
                        "width": 50,
                      }
                    }
                  />
                </View>
                <View
                  accessible={true}
                  isTVSelectable={true}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "margin": "5%",
                      "opacity": 1,
                    }
                  }
                >
                  <Image
                    resizeMode="contain"
                    source={1}
                    style={
                      Object {
                        "height": 50,
                        "width": 50,
                      }
                    }
                  />
                </View>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "marginLeft": 15,
                      "marginRight": 15,
                    },
                    undefined,
                    undefined,
                    undefined,
                  ]
                }
              >
                <View
                  accessible={true}
                  isTVSelectable={true}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={null}
                >
                  <View
                    pointerEvents="box-only"
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#9E9E9E",
                          "flexDirection": "row",
                          "justifyContent": "center",
                          "padding": 19,
                        },
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        Object {
                          "padding": 12,
                        },
                        undefined,
                        undefined,
                        undefined,
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#2A5E85",
                          "borderColor": "white",
                          "borderRadius": 30,
                          "height": 50,
                          "justifyContent": "center",
                          "paddingLeft": 10,
                          "width": 180,
                        },
                        undefined,
                        undefined,
                      ]
                    }
                  >
                    <Text />
                    <Text
                      style={
                        Array [
                          Object {},
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          Array [
                            Object {
                              "color": "white",
                              "fontSize": 20,
                            },
                            undefined,
                            Object {
                              "fontSize": 17.5,
                            },
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                          ],
                        ]
                      }
                    >
                      RETOUR
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </Modal>
      </View>
    </View>
  </View>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    supportedOrientations={
      Array [
        "portrait",
        "portrait-upside-down",
        "landscape",
        "landscape-left",
        "landscape-right",
      ]
    }
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flex": 1,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#FFF",
            "borderColor": "#2A5E85",
            "borderRadius": 5,
            "borderWidth": 1,
            "color": "#2A5E85",
            "marginHorizontal": 30,
            "maxWidth": 350,
            "shadowColor": "#000000",
            "shadowOpacity": 0.1,
            "shadowRadius": 10,
            "transform": Array [
              Object {
                "scale": 0,
              },
            ],
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "justifyContent": "center",
                "padding": 20,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#333",
                    "fontSize": 18,
                    "fontWeight": "500",
                    "lineHeight": 25,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "red",
                  },
                ]
              }
            >
              ATTENTION
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#666",
                    "fontSize": 17,
                    "paddingTop": 10,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#2A5E85",
                  },
                ]
              }
            >
              Vous n'avez plus assez de crédit gratuit, pour acheter cette application, rendez-vous sur l'application web pour choisir un autre moyen de paiement.
            </Text>
          </View>
          <View
            style={
              Object {
                "borderRadius": 5,
                "flexDirection": "row",
                "flexWrap": "wrap",
                "justifyContent": "center",
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#2A5E85",
                  "borderRadius": 5,
                  "marginBottom": 20,
                  "marginHorizontal": 10,
                  "minWidth": 90,
                  "opacity": 1,
                  "paddingVertical": 8,
                }
              }
              testID="buttonConfirm"
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#FFF",
                      "fontSize": 15,
                      "fontWeight": "600",
                      "textAlign": "center",
                    },
                    undefined,
                  ]
                }
              >
                RETOUR
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
